File: src/main/java/io/curiousoft/ijudi/ordermanagement/repo/OrderRepository.java
package io.curiousoft.ijudi.ordermanagement.repo;

import io.curiousoft.ijudi.ordermanagement.model.Order;
import io.curiousoft.ijudi.ordermanagement.model.OrderStage;
import io.curiousoft.ijudi.ordermanagement.model.ShippingData;
import org.springframework.data.mongodb.repository.MongoRepository;

import java.util.Date;
import java.util.List;
import java.util.Optional;

public interface OrderRepository extends MongoRepository<Order, String> {

    Optional<List<Order>> findByCustomerId(String customerId);

    List<Order> findByShopId(String id);

    List<Order> findByShopIdAndStageNot(String id, OrderStage stage);

    void deleteByShopPaidAndStageAndModifiedDateBefore(boolean shopPaid, OrderStage stage, Date date);

    List<Order> findByShippingDataMessengerIdAndStageNot(String id, OrderStage customerNotPaid);

    List<Order> findByStage(OrderStage eq);

    List<Order> findByShopPaidAndStageAndModifiedDateBefore(boolean b, OrderStage stage6WithCustomer, Date pastDate);

    List<Order> findByMessengerPaidAndStageAndShippingData_Type(boolean paid, OrderStage stage6WithCustomer, ShippingData.ShippingType delivery);
}

File: izinga-commons/src/main/kotlin/io/curiousoft/izinga/commons/repo/OrderRepository.kt
package io.curiousoft.izinga.commons.repo

import io.curiousoft.izinga.commons.model.Order
import io.curiousoft.izinga.commons.model.OrderStage
import io.curiousoft.izinga.commons.model.ShippingData
import org.springframework.data.mongodb.repository.MongoRepository
import java.util.*
import javax.validation.constraints.NotNull

interface OrderRepository : MongoRepository<Order?, String?> {
    fun findByCustomerId(customerId: String?): Optional<List<Order?>?>?
    fun findByShopId(id: String?): List<Order?>?
    fun findByShopIdAndStageNot(id: String?, stage: OrderStage?): List<Order?>?
    fun deleteByShopPaidAndStageAndModifiedDateBefore(shopPaid: Boolean, stage: OrderStage?, date: Date?)
    fun findByShippingDataMessengerIdAndStageNot(id: String?, customerNotPaid: OrderStage?): List<Order?>?
    fun findByStage(eq: OrderStage?): List<Order?>?
    fun findByShopPaidAndStageAndModifiedDateBefore(
        b: Boolean,
        stage6WithCustomer: OrderStage?,
        pastDate: Date?
    ): List<Order?>?

    fun findByMessengerPaidAndStageAndShippingData_Type(
        paid: Boolean,
        stage6WithCustomer: OrderStage?,
        delivery: ShippingData.ShippingType?
    ): List<Order?>?

    fun findByShopPaidAndStage(paid: Boolean, stage7AllPaid: @NotNull OrderStage?): List<Order>?
    fun findByMessengerPaidAndStage(paid: Boolean, stage7AllPaid: @NotNull OrderStage?): List<Order>?
    fun findByIdIn(orderIds: List<String>): List<Order>
    fun findByCustomerIdAndShippingDataMessengerIdAndStageIn(customerId: String, messengerId: String, stages: Array<OrderStage>): List<Order>?
}